File: src/main/java/me/kaotich00/fwauctionhouse/commands/user/ValidateTokenCommand.java
package me.kaotich00.fwauctionhouse.commands.user;

import me.kaotich00.fwauctionhouse.commands.api.UserCommand;
import me.kaotich00.fwauctionhouse.message.Message;
import me.kaotich00.fwauctionhouse.objects.PendingToken;
import me.kaotich00.fwauctionhouse.services.SimpleMarketService;
import me.kaotich00.fwauctionhouse.storage.StorageFactory;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Optional;

public class ValidateTokenCommand extends UserCommand {

    @Override
    public void onCommand(CommandSender sender, String[] args) {
        super.onCommand(sender, args);

        String sessionId = args[1];

        Optional<PendingToken> optPendingToken = SimpleMarketService.getInstance().getPendingToken(Integer.parseInt(sessionId));
        if(optPendingToken.isPresent()) {
            PendingToken pendingSell = optPendingToken.get();
            Message.VALIDATED_TOKEN.send(sender);
            StorageFactory.getInstance().getStorageMethod().validateToken(Integer.parseInt(sessionId));
            SimpleMarketService.getInstance().removeFromPendingToken(pendingSell);
        }

    }

    @Override
    public String getInfo() {
        return super.getInfo();
    }

    @Override
    public String getUsage() {
        return "/market validateToken <sessionId>";
    }

    @Override
    public String getName() {
        return super.getName();
    }

    @Override
    public Integer getRequiredArgs() {
        return 1;
    }

}

File: src/main/java/me/kaotich00/fwauctionhouse/commands/user/ValidateTokenCommand.kt
package me.kaotich00.fwauctionhouse.commands.user

import com.google.inject.Inject
import me.kaotich00.fwauctionhouse.commands.api.UserCommand
import me.kaotich00.fwauctionhouse.message.Message
import me.kaotich00.fwauctionhouse.services.ListingsService
import me.kaotich00.fwauctionhouse.storage.ListingsDao
import org.bukkit.entity.Player

class ValidateTokenCommand @Inject constructor(
    private val listingsService: ListingsService,
    private val listingsDao: ListingsDao
) : UserCommand(
    "validateToken",
    "",
    1,
    "/market validateToken <sessionId>"
) {

    override fun doCommand(sender: Player, args: Array<String>) {
        val sessionId = args[1].toIntOrNull() ?: run {
            sender.sendMessage("You must insert a valid session id")
            return
        }

        val pendingToken = listingsService.getPendingToken(sessionId) ?: return

        Message.VALIDATED_TOKEN.send(sender)
        listingsDao.validateToken(sessionId)
        listingsService.removeFromPendingToken(pendingToken)
    }

}